### PR TITLE
Change styling of TEC member-side dashboard 

### PR DESCRIPTION
**Pull request by airaagrawal1**
### Summary 

Changed styling of TEC member-side dashboard to card format

### Test Plan 

Added more data to dev firebase and made sure that logic was being displaying correctly. Here is a screenshot of the current formatting:
Screen Shot 2022-03-15 at 12 42 25 PM

### CODE REVIEW

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx
Comment (henry-li-06) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx:226]: I think displaying the number of credits would also be good

**Review by henry-li-06 — COMMENTED** [review]
The dashboard should probably be refactored into a separate component, there's a lot of logic/computation that's completely unrelated to the rest of the form.

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx
Comment (andrew032011) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx:241]: Small typo here. Pending Approval For:: -> Pending Approval For:, or probably just "Pending Approval:"

**Review by andrew032011 — COMMENTED** [review]
I was also wondering your thoughts on putting a count for the number of pending credits? Might make it easier for user to see when they've reached 3 (even if it's unapproved). 

This might be a separate PR, but I also wonder how events are handled if they end up not being approved.

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx
Comment (airaagrawal1) [reply frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx]: Good idea!

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx
Comment (airaagrawal1) [reply frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx]: > I was also wondering your thoughts on putting a count for the number of pending credits? Might make it easier for user to see when they've reached 3 (even if it's unapproved).
> 
> This might be a separate PR, but I also wonder how events are handled if they end up not being approved.

I think Henry's comment of adding credit counts to each Team Event card should handle showing them how many credits they're waiting to be approved for. 

Good question about how non-approved events are handled! @henry-li-06 what do you think?

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventsCreditsDasboard.tsx
Comment (henry-li-06) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventsCreditsDasboard.tsx:8]: use the `useSelf` hoook

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventsCreditsDasboard.tsx
Comment (henry-li-06) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventsCreditsDasboard.tsx:47]: move the inline styling to a separate `.module.css` file

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventsCreditsDasboard.tsx
Comment (henry-li-06) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventsCreditsDasboard.tsx:54]: inline styling ^

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventsCreditsDasboard.tsx
Comment (henry-li-06) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventsCreditsDasboard.tsx:55]: inline styling

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx
Comment (henry-li-06) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx:165]: inline styling

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx
Comment (henry-li-06) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx:150]: inline styling

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx
Comment (henry-li-06) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx:92]: inline styling

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventsCreditsDasboard.tsx
Comment (henry-li-06) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventsCreditsDasboard.tsx:60]: inline styling

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventsCreditsDasboard.tsx
Comment (henry-li-06) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventsCreditsDasboard.tsx:84]: inline styling

**Comment by dan-ieljin** [comment]
The card styling looks great! 

What does the page look like with no approved/pending events, and with a second row of events? I agree with Henry that displaying the # of credits on the cards themselves would be a good idea as a reminder for the user. Also, would the label "Remaining Credits Needed" look cleaner alongside the "Approved Credits" label to reduce the number of lines? Or is it more intuitive the current layout?

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx
Comment (henry-li-06) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx:152]: move inline styling

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx
Comment (henry-li-06) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx:168]: inline styling

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx
Comment (henry-li-06) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx:170]: inline styling

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventsCreditDasboard.tsx
Comment (henry-li-06) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventsCreditDasboard.tsx:16]: get rid of commented out code

**Comment by airaagrawal1** [comment]
> The card styling looks great!
> 
> What does the page look like with no approved/pending events, and with a second row of events? I agree with Henry that displaying the # of credits on the cards themselves would be a good idea as a reminder for the user. Also, would the label "Remaining Credits Needed" look cleaner alongside the "Approved Credits" label to reduce the number of lines? Or is it more intuitive the current layout?

With no approved/pending events, there is a message that displays that the member does not have any events in that category. I think I broke up the number of credits needed and approved so that the corresponding TEC list would be right underneath it. We can certainly change the design if you think the other way would be better tho!

**Review by henry-li-06 — APPROVED** [review]
Nice work, LGTM!
```
              |    |    |                 
             )_)  )_)  )_)              
            )___))___))___)\            
           )____)____)_____)\\
         _____|____|____|____\\\__
---------\                   /---------
  ^^^^^ ^^^^^^^^^^^^^^^^^^^^^
    ^^^^      ^^^^     ^^^    ^^
         ^^^^      ^^^
```

**Pull request merged**
